plugins/node/opentelemetry-instrumentation-express/test/express.test.ts

```diff
@@ -275,6 +275,35 @@ describe('ExpressInstrumentation', () => {
       assert.strictEqual(memoryExporter.getFinishedSpans().length, 0);
       assert.strictEqual(res, 'test');
     });
+
+    it('should update rpcMetadata.route with the latest middleware layer', async () => {
+      const rootSpan = tracer.startSpan('rootSpan');
+      let rpcMetadata: RPCMetadata | undefined;
+      const httpServer = await serverWithMiddleware(tracer, rootSpan, app => {
+        app.use(express.json());
+        app.use((req, res, next) => {
+          rpcMetadata = getRPCMetadata(context.active());
+          next();
+        });
+
+        app.use('/bare_middleware', (req, res) => {
```

Comment:
This test should assert that the route is still captured when 
```js
type === ExpressLayerType.REQUEST_HANDLER
```

This type will be set when `layer.name === 'bound dispatch'`
```js
  } else if (layer.name === 'bound dispatch') {
    return {
      attributes: {
        [AttributeNames.EXPRESS_NAME]: layerPath ?? 'request handler',
        [AttributeNames.EXPRESS_TYPE]: ExpressLayerType.REQUEST_HANDLER,
      },
      name: `request handler${layer.path ? ` - ${layerPath}` : ''}`,
    };
```

Is using the `app.use` triggers a layer with name 'bound dispatch'?

Reply:
```
  const router = express.Router();

  router.use("/router_use", () => {}); // #3
  router.get('/tada/, () => {}); // #2

  app.use('/toto', router); // #1
  app.use('/bare_use', () => {}); // #3

  app.get('/bare_get', () => {}); // #2
```

"#1", the middleware that bound app.use/router.use with Router object is 'router' layer

"#2" is `bound dispatch`, ie, route handler with HTTP method name. This is for both Express's App and Express's Router object. This is something our unit test cover alot.

"#3" is the last case, IE middleware layer, where app.use/router.use have generic handler. It can be with or without the route's parameter

Most of our test with app.use/router.use is for common's middleware use case, that don't have layername at the begining.

This new test for specificly app.use with route layer.

Reply:
I add additional test for these 2 case, as I don't see we have specific test for them
```
  router.use("/router_use", () => {}); // #3
  app.get('/bare_get', () => {}); // #2
```
internally I think they are the same with case that we already cover but just to make sure.